apps/emqx_eviction_agent/test/emqx_eviction_agent_SUITE.erl

```diff
@@ -0,0 +1,232 @@
+end_per_testcase(_TestCase, _Config) ->
+    _ = emqx_eviction_agent:disable(test_eviction).
+
+t_enable_disable(_Config) ->
+    erlang:process_flag(trap_exit, true),
+
+    ?assertMatch(
+       disabled,
+       emqx_eviction_agent:status()),
+
+    {ok, C0} = emqtt_connect(),
+    ok = emqtt:disconnect(C0),
+
+    ok = emqx_eviction_agent:enable(test_eviction, undefined),
+
+    ?assertMatch(
+       {error, eviction_agent_busy},
+       emqx_eviction_agent:enable(bar, undefined)),
+
+    ?assertMatch(
+       ok,
+       emqx_eviction_agent:enable(test_eviction, <<"srv">>)),
+
+    ?assertMatch(
+        {enabled, #{}},
+        emqx_eviction_agent:status()),
+
+    ?assertMatch(
+       {error, {use_another_server, #{}}},
+       emqtt_connect()),
+
+    ?assertMatch(
+        {error, eviction_agent_busy},
+        emqx_eviction_agent:disable(bar)),
+
+    ?assertMatch(
+        ok,
+        emqx_eviction_agent:disable(test_eviction)),
+
+    ?assertMatch(
+        {error, disabled},
+        emqx_eviction_agent:disable(test_eviction)),
+
+    ?assertMatch(
+       disabled,
+       emqx_eviction_agent:status()),
+
+    {ok, C1} = emqtt_connect(),
+    ok = emqtt:disconnect(C1).
+
+
+t_evict_connections_status(_Config) ->
+    erlang:process_flag(trap_exit, true),
+
+    {ok, _C} = emqtt_connect(),
+
+    {error, disabled} = emqx_eviction_agent:evict_connections(1),
+
+    ok = emqx_eviction_agent:enable(test_eviction, undefined),
+
+    ?assertMatch(
+       {enabled, #{connections := 1, sessions := _}},
+       emqx_eviction_agent:status()),
+
+    ok = emqx_eviction_agent:evict_connections(1),
+
+    ct:sleep(100),
+
+    ?assertMatch(
+       {enabled, #{connections := 0, sessions := _}},
+       emqx_eviction_agent:status()),
+
+    ok = emqx_eviction_agent:disable(test_eviction).
+
+
+t_explicit_session_takeover(Config) ->
+    erlang:process_flag(trap_exit, true),
+
+    {ok, C0} = emqtt_connect(<<"client_with_session">>, false),
+    {ok, _, _} = emqtt:subscribe(C0, <<"t1">>),
+
+    ok = emqx_eviction_agent:enable(test_eviction, undefined),
+
+    ?assertEqual(
+       1,
+       emqx_eviction_agent:connection_count()),
+
+    ok = emqx_eviction_agent:evict_connections(1),
+
+    receive
+        {'EXIT', C0, {disconnected, ?RC_USE_ANOTHER_SERVER, _}} -> ok
+    after 1000 ->
+              ?assert(false, "Connection not evicted")
+    end,
+
+    ?assertEqual(
+       0,
+       emqx_eviction_agent:connection_count()),
+
+    ?assertEqual(
+       1,
+       emqx_eviction_agent:session_count()),
+
+    %% First, evacuate to the same node
+
+    ?check_trace(
+       ?wait_async_action(
+          emqx_eviction_agent:evict_sessions(1, node()),
+          #{?snk_kind := emqx_channel_takeover_end},
+          1000),
+       fun(_Result, Trace) ->
+        ?assertMatch(
+           [#{clientid := <<"client_with_session">>} | _ ],
+           ?of_kind(emqx_channel_takeover_end, Trace))
+       end),
```

Comment:
Q: This shouldn't be allowed, if the node is being evacuated?

Reply:
This handled in the rebalancing app now.